### ISSUE TITLE
prefer-stateless-function should check if the component depends on the Context API

### ISSUE THREAD

**Issue by bensampaio** [yannickcr/eslint-plugin-react#1419]
My component is as follows:

```
class MyComponent extends Component {
  render() {
    const { someValue } = this.context;

   return <div>{someValue}</div>;
  }
}
```

If I am not wrong it is not possible to access the Context property on a stateless component so the rule should recognise this case as valid.

**Comment by jseminck** [comment]
It seems like this is according to design:

Here is a test case covering it:

https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/prefer-stateless-function.js#L460

According to the docs of React, the context is passed in as the second parameter of the stateless function: https://facebook.github.io/react/docs/context.html#referencing-context-in-stateless-functional-components

So your example would be:

```js
function MyComponent(props, context) {
   return <div>{context.someValue}</div>
}
```

or ...

```js
function MyComponent(props, { somevalue }) {
   return <div>{someValue}</div>
}
```
